### PR TITLE
Split config handlers for better customization

### PR DESCRIPTION
**Pull request by hkupty**
This implementation allows for better utilization of the config class as it can be specialized for the different sources `java-language-server` supports.

For Bazel, it also allows for switching the binary, so the server can be configured to used a different version of the binary or use `bazelisk` as frontend.

Finally, as it was difficult to debug without accessing the logs, this PR also introduces parsing the CLI flags to enable logging to a file.

### CODE REVIEW

File: src/main/java/org/javacs/config/MavenConfig.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
+package org.javacs.config;
```
Comment (hkupty) [inline src/main/java/org/javacs/config/MavenConfig.java:1]: This is the same case as `BazelConfig`, except this is still not being used. Couldn't really think on the better logic for picking this one over `InferConfig`, so I decided not to hook it on this PR.

**Comment by georgewfraser** [comment]
Does this change behavior, or is this strictly a refactor?

**Comment by hkupty** [comment]
There are three things embedded in this PR:
- Parsing command-line arguments;
- Splitting the config into multiple, specialized files;
- Allowing bazel binary to be set via config.

I have to use `bazelisk` as a frontend for `bazel` due to version difference between the project and the bundled version of the OS.
In order for me to do that, I had to go down the `InferConfig` class and understand how things are managed there.

What follows however is that it will try to infer the config at least twice, once for classpath and once for docpath. Also, the class has some shared functions that could be reused while others are single-use for one implementation (i.e. `fork` for `bazel` only or `mvnDependencies` for maven);

I can undo the refactoring, though I see value in splitting the config mechanisms. WDYT?

**Comment by georgewfraser** [comment]
I'm sort of anti-OOP. I prefer to use ordinary functions calling each other statically whenever possible. 

Let me ask a different question: what are the behavior changes in this PR?

**Comment by hkupty** [comment]
So, sorry for taking so long to get back to this. I'm inclined to drop this and come up with a simpler one that simply allows me to set the bazel binary I want through config (in this case, `bazelisk`), if that's not possible already.

**Pull request closed without merging**